cmake/sca/eclair/sca.cmake

```diff
@@ -0,0 +1,140 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2023, BUGSENG Srl
+
+find_program(ECLAIR_ENV eclair_env REQUIRED)
+message(STATUS "Found eclair_env: ${ECLAIR_ENV}")
+
+find_program(ECLAIR_REPORT eclair_report REQUIRED)
+message(STATUS "Found eclair_report: ${ECLAIR_REPORT}")
+
+# ECLAIR Settings
+set(ECLAIR_PROJECT_NAME "Zephyr-${BOARD}${BOARD_QUALIFIERS}")
+set(ECLAIR_OUTPUT_DIR ${CMAKE_BINARY_DIR}/sca/eclair_out)
+set(ECLAIR_ECL_DIR "${ZEPHYR_BASE}/cmake/sca/eclair/ECL")
+set(ECLAIR_ANALYSIS_ECL_DIR "${ZEPHYR_BASE}/cmake/sca/eclair/ECL")
+set(ECLAIR_DIAGNOSTICS_OUTPUT "${ECLAIR_OUTPUT_DIR}/DIAGNOSTIC.txt")
+set(ECLAIR_DATA_DIR "${ECLAIR_OUTPUT_DIR}/data")
+set(ECLAIR_BUILD_DIR "${ECLAIR_OUTPUT_DIR}/build")
+set(ECLAIR_PROJECT_ECD "${ECLAIR_OUTPUT_DIR}/PROJECT.ecd")
+set(CC_ALIASES "${CMAKE_C_COMPILER}")
+set(CXX_ALIASES "${CMAKE_CXX_COMPILER}")
+set(AS_ALIASES "${CMAKE_AS}")
+set(LD_ALIASES "${CMAKE_LINKER}")
+set(AR_ALIASES "${CMAKE_ASM_COMPILER_AR} ${CMAKE_C_COMPILER_AR} ${CMAKE_CXX_COMPILER_AR}")
+
+set(ECLAIR_ENV_ADDITIONAL_OPTIONS "")
+set(ECLAIR_REPORT_ADDITIONAL_OPTIONS "")
+
+# Default value
+set(ECLAIR_RULESET first_analysis)
+
+# ECLAIR env
+if (CONFIG_ECLAIR_RULESET_FIRST_ANALYSIS)
+  set(ECLAIR_RULESET first_analysis)
+endif()
+if (CONFIG_ECLAIR_RULESET_STU)
+  set(ECLAIR_RULESET STU)
+endif()
+if (CONFIG_ECLAIR_RULESET_STU_HEAVY)
+  set(ECLAIR_RULESET STU_heavy)
+endif()
+if (CONFIG_ECLAIR_RULESET_WP)
+  set(ECLAIR_RULESET WP)
+endif()
+if (CONFIG_ECLAIR_RULESET_STD_LIB)
+  set(ECLAIR_RULESET std_lib)
+endif()
+if (CONFIG_ECLAIR_RULESET_USER)
+  set(ECLAIR_RULESET ${CONFIG_ECLAIR_USER_RULESET_NAME})
+  if(IS_ABSOLUTE ${CONFIG_ECLAIR_USER_RULESET_PATH})
+      set(ECLAIR_ANALYSIS_ECL_DIR ${CONFIG_ECLAIR_USER_RULESET_PATH})
+  else()
+    set(ECLAIR_ANALYSIS_ECL_DIR ${APPLICATION_SOURCE_DIR}/${CONFIG_ECLAIR_USER_RULESET_PATH})
```

Comment:
why is `APPLICATION_SOURCE_DIR` chosen here ?

Generally `APPLICATION_CONFIG_DIR` is more appropriate so that if follows same behavior as rest of build system.
https://docs.zephyrproject.org/latest/develop/application/index.html#application-configuration-directory

Note, that default value of `APPLICATION_CONFIG_DIR` is the value of `APPLICATION_SOURCE_DIR`